src/app/static/src/app/store/modelCalibrate/actions.ts

```diff
@@ -124,6 +120,39 @@ export const actions: ActionTree<ModelCalibrateState, RootState> & ModelCalibrat
         if (state.calibrating && !state.complete && state.calibrateId) {
             await dispatch("poll");
         }
+    },
+
+    async getResultData(context, indicatorId) {
+        const {commit, state} = context;
+        const calibrateId = state.calibrateId;
+
+        if (!state.status.done || !indicatorId) {
+            // Don't try to fetch data if the calibration hasn't finished
+            return
+        }
+
+        let indicatorKnown = false;
+        if (state.result) {
+            indicatorKnown = state.result.data.some(row => {
+                return row.indicator === indicatorId
```

Comment:
as we discussed earlier you could have an array in state that stores which indicators are fetched and then have a mutation that pushes to the data array and pushes to the indicatorsFetched array to make sure they are in sync but that can be done in the next iteration of this!

Reply:
Thanks yep - I've put this in as seems worth doing here and wasn't too bad in the end.